### ISSUE TITLE
Mouse bindings

### ISSUE THREAD

**Issue by The-Compiler** [The-Compiler/qutebrowser#591]
The idea just came up in IRC - it might or might not be a good idea to be able to bind mouse keypresses (with optional modifiers) in the config.

**Comment by Ram-Z** [comment]
This would allow us to map `open -b {url}` to `Shift+MiddleClick`.

**Comment by The-Compiler** [comment]
It might also be a good idea to allow binding wheel events in some way, i.e. to reverse scrolling, or change the speed.

**Comment by NoctuaNivalis** [comment]
Or scroll through time :astonished:

**Comment by courtyardz** [comment]
I just found out about how `shift + scrollwheel` enables horizontal scrolling... :)

This is just a bump, but...
 
Definitely would be really cool to map, among other things, the left and right clicks on my mouse's scrollwheel to go back/forward in history (for times when one is just mousing around)

**Comment by R3load** [comment]
If you're going to support rebinding keys, please support *all* mouse buttons *and* the scroll wheel.
This is a plague even in games, take FEAR2 for example. Can't really bind thumb buttons there.
Consistency, consistency, consistency. Buttons on the mouse *are* buttons. And so are scroll wheel events. On Linux (at least with libinput), these are just another button in fact.

**Comment by prosoitos** [comment]
@courtyardz `ctrl + scrollwheel` allows to play with the zoom too.

Very interested in this as well (though I only wish to do minor changes to the modifiers during mouse clicks).

**Comment by nancym** [comment]
If you're on Windows, this is useful:

> X-Mouse Button Control is a free tool for Windows that allows you to re-configure and expand the capabilities of your mouse.
> http://www.highrez.co.uk/downloads/XMouseButtonControl.htm

I :heart: XMBC (and qutebrowser too)!

**Comment by courtyardz** [comment]
If you're on Linux, A.B.'s answer at

https://askubuntu.com/questions/236529/binding-back-forward-to-mouse-buttons#634239

let me just start throwing ALT + LEFT- or RIGHT-ARROWS key presses at
Firefox with left or right tilts of my scrollwheel. Firefox interprets
those as Back or Forward, so then one gets v effortless, mouse-only history
navigation! Should work too on Qtbrowser...  Truly a revelation for me,
hope this gets folded in to the project one day

Thank you for tolerating my noob mouse geek-out!

**Comment by R3load** [comment]
Also, we could use drag-scrolling done by holding down a key (or mouse button, I'd use RMB/Mouse2/button3 personally) and dragging the cursor on the page in a direction opposite to the scrolling direction.

**Comment by linkert** [comment]
I'd like to switch between tabs with Shift+wheel{Up,Down}, went on to bind this but turns out it's not yet possible to bind actions to the scrollwheel?

**Comment by The-Compiler** [comment]
@linkert Correct. You can use the mouse wheel on the tab bar to switch tabs, though.

**Comment by linkert** [comment]
> @linkert Correct. You can use the mouse wheel on the tab bar to switch tabs, though.

Tab bar is hidden ;) Would like to do the Shift+Scrollwheel switching without the tab bar showing momentarily (I've got it set to 500 milliseconds). Like a "*quiet switch*" between tabs. That would be awesome since I'm still not keen on the tab bar "panel" pushing, "squeezing" or rather resizing the content/site while the tab bar is showing only to resize again when it hides, couldn't it just float on top of the site/content? - a bit of a different issue though.

**Comment by The-Compiler** [comment]
Another consideration, [from Reddit](https://www.reddit.com/r/qutebrowser/comments/s7pnk8/can_i_bind_clicking_on_the_tab_bar_to_execute/), is the ability to bind clicks to the empty area of the tabbar:

> Basically I have a vertical tabbar which is normally thin and shows only favicons. I wanted clicking the tab bar to expand it such that it shows the title as well, (increase width and change tab title format).

**Comment by shiar** [comment]
Could also allow mouse buttons as modifiers, like `:bind <RightClick+WheelUp> scroll-page 0 -1` to scroll faster while holding the right button. I'm currently running _easystroke_ on Linux for this feature, but would love built-in support.

**Comment by prosoitos** [comment]
I have a new use-case and I hope that restarting this conversation will revive interest in this 🙂 

I got a new monitor and I am having terrible VA smearing when scrolling (I use dark mode + redshift and that probably doesn't help).

This happens when scrolling with the mouse or when using commands such as `scroll up` or `scroll down`. However, I don't get any of that if I use `scroll-page 0 0.01` and `scroll-page 0 -0.01` instead.

So I would love to associate the mouse scrolling to `scroll-page 0 0.01` and `scroll-page 0 -0.01` instead of actual scrolling. Maybe that would lead to a jittery effect, but I don't care about that. I could try even smaller increments. And it could only be better than what I am seeing right now.
